repository_service_tuf/cli/admin/ceremony.py

```diff
@@ -229,219 +237,438 @@
             console.print(":cross_mark: [red]Failed[/]: Key is duplicated.")
             continue
 
-        role.keys[f"{rolename}_{key_count}"] = {
-            "filename": filepath.split("/")[-1],
-            "password": password,
-            "key": key.key,
-        }
+        yield role_key
+
         console.print(
             ":white_check_mark: Key "
-            f"{key_count}/{role.num_of_keys} [green]Verified[/]"
+            f"{key_count}/{number_of_keys} [green]Verified[/]"
         )
         key_count += 1
 
 
-def _bootstrap(server, headers, json_payload) -> Optional[str]:
-    task_id = None
-    response = request_server(
-        server,
-        URL.bootstrap.value,
-        Methods.post,
-        json_payload,
-        headers=headers,
-    )
-    response_json = response.json()
-    if response.status_code != 202:
-        raise click.ClickException(
-            f"Error {response.status_code} {response_json.get('detail')}"
+def _run_user_validation():
+    # Tables
+    #
+    # The table.Table doesn't have a library API to remove/clean the rows, so
+    # we call the function to reset/construct it.
+    def _init_summary_table(title: str) -> table.Table:
+        online_key_table = table.Table()
+        online_key_table.add_column(
+            f"{title}",
+            justify="left",
+            vertical="middle",
+        )
+        return online_key_table
+
+    def _init_keys_table(path: Optional[bool] = True) -> table.Table:
+        """Gets a new keys table"""
+        keys_table = table.Table(box=box.MINIMAL)
+        if path is True:
+            keys_table.add_column(
+                "path", justify="right", style="cyan", no_wrap=True
+            )
+        keys_table.add_column("type", justify="center")
+        keys_table.add_column("verified", justify="center")
+        keys_table.add_column("id", justify="center")
+
+        return keys_table
+
+    # Validations
+    #
+    # Online Key validation
+    while True:
+        online_key_table = _init_summary_table("ONLINE KEY SUMMARY")
+        keys_table = _init_keys_table()
+        keys_table.add_row(
+            f"[yellow]{setup.online_key.key_path}[/]",
+            "[green]Online[/]",
+            ":white_heavy_check_mark:",
+            f"[yellow]{setup.online_key.key.get('keyid')}[/]",
         )
 
-    elif (
-        response_json.get("message") is None
-        or response_json.get("message") != "Bootstrap accepted."
-    ):
-        raise click.ClickException(response.text)
+        online_key_table.add_row(keys_table)
+        console.print("\n", online_key_table)
 
-    else:
-        if data := response_json.get("data"):
-            task_id = data.get("task_id")
-            console.print(f"Bootstrap status: ACCEPTED ({task_id})")
+        confirm_config = prompt.Confirm.ask(
+            "\nIs [cyan]Online Key[/] configuration correct?"
+        )
+        if not confirm_config:
```

Comment:
Yes, it is a safety check because we use the `prompt.Config.ask`.
https://github.com/vmware/repository-service-tuf-cli/blob/e8e30ba510c66849ebe2432992d9cc9b21045b9b/repository_service_tuf/cli/admin/ceremony.py#L524
[By default it returns a boolean](https://rich.readthedocs.io/en/stable/reference/prompt.html#rich.prompt.Confirm)
It makes to avoid using choices, by default implements choices as `y/n`

Reply:
I made it more explicit in my last commit.